### PR TITLE
Add LOG_LEVEL & supporting log functions

### PR DESCRIPTION
**Pull request by martin-walsh**
Issue Ref: https://github.com/veqryn/kube-cloudwatch-autoscaler/issues/9

Adds ERROR (red prefix), INFO (green prefix), DEBUG (yellow prefix) functions controlled by LOG_LEVEL env var.

### CODE REVIEW

**Comment by martin-walsh** [comment]
Will try get a test run built.

**Comment by martin-walsh** [comment]
image
Local test run output.

**Comment by martin-walsh** [comment]
Test run with LOG_LEVEL=ERROR

image

File: entrypoint.sh [hunk 1]
```diff
@@ -30,6 +30,33 @@ CW_DIMENSIONS_DELIMITER="${CW_DIMENSIONS_DELIMITER:-" "}"
...
+log_info() {
+  if [[ "${LOG_LEVEL}" == "INFO" || "${LOG_LEVEL}" == "DEBUG" ]]; then
+    echo -e "\033[1;32m$(date -u -I'seconds') INFO:\033[0m ${1}"
+  fi
+}
+
+log_debug() {
+  # Maintain backwards compat for VERBOSE flag
+  if [[ "${LOG_LEVEL}" == "DEBUG" || "${VERBOSE}" == true ]]; then
+    echo -e "\033[1;33m$(date -u -I'seconds') DEBUG:\033[0m ${1}"
```
Comment (veqryn) [inline entrypoint.sh]: I usually associate yellow with warnings. Maybe we should keep debug as white?

File: README.md [hunk 2]
```diff
@@ -65,6 +65,8 @@ spec:
               value: "30"
             - name: VERBOSE # Optional, will log kube and cloudwatch statistics
               value: "false"
+            - name: LOG_LEVEL # Optional, defaults to "INFO". Allowable values: "DEBUG", "INFO", "ERROR", "OFF"
```
Comment (veqryn) [inline README.md]: should we remove the VERBOSE flag, since it is superseded by DEBUG level?

**Comment by veqryn** [comment]
I assume you are actively using this in production, correct?
If so, can you please have a look at the other 1 issue and other 1 PR, and see we should fix or merge those?  
(I was previously using this project in production, but stopped when that project got shut down, so I am hesitant to make changes without being able to thoroughly test them.)

File: entrypoint.sh (hunk 1 above)
Comment (martin-walsh) [reply entrypoint.sh]: Will do.

File: README.md (hunk 2 above)
Comment (martin-walsh) [reply README.md]: It would be cleaner, but I was attempting to keep backward compat. I have no objection to dropping it.

**Comment by martin-walsh** [comment]
> I assume you are actively using this in production, correct? If so, can you please have a look at the other 1 issue and other 1 PR, and see we should fix or merge those? (I was previously using this project in production, but stopped when that project got shut down, so I am hesitant to make changes without being able to thoroughly test them.)

I had a look at the other PR, not exactly sure why you would want to pass through a raw CW value e.g. scale to the depth of an SQS queue with 1,000,000 items. Given it's been 5 years, we could reach out but I reckon that the requester has either moved on or chosen a different path.

Re the `trap` issue, fixed by passing through exit code.

If you are no longer using this in production, are you happy to keep maintaining it going forward ?

File: entrypoint.sh (hunk 1 above)
Comment (martin-walsh) [reply entrypoint.sh]: Done

File: README.md (hunk 2 above)
Comment (martin-walsh) [reply README.md]: Done. It's pretty easy to set a log level.

File: example_autoscaler_kube.yaml [hunk 3]
```diff
@@ -52,6 +52,8 @@ spec:
               value: "30"
             - name: VERBOSE # Optional, will log kube and cloudwatch statistics
               value: "false"
+            - name: LOG_LEVEL # Optional, defaults to "INFO". Allowable values: "DEBUG" (will log kube and cloudwatch statistics), "INFO" (will log scaling activity), "ERROR" (will log errors only), "OFF" (nothing)
```
Comment (veqryn) [inline example_autoscaler_kube.yaml:53]: can remove verbose here too ^

**Review by veqryn — APPROVED** [review]
Let me know when you are ready for a merge

**Comment by veqryn** [comment]
I could add you to the project, if you'd like to help maintain it.

File: example_autoscaler_kube.yaml (hunk 3 above)
Comment (martin-walsh) [reply example_autoscaler_kube.yaml:53]: Done

**Comment by martin-walsh** [comment]
> Let me know when you are ready for a merge

Ready.

> I could add you to the project, if you'd like to help maintain it.

Up to you, I'm happy to pitch in.

**Review by veqryn — APPROVED** [review]

**Comment by veqryn** [comment]
@martin-walsh merged and added you as a collaborator. Let me know if I need to change any other settings for you.

**Pull request merged**